### ISSUE TITLE
useClass ignores top-level name property

### ISSUE THREAD

**Issue by solshark** [nestjs/bull#285]
## I'm submitting a...

<pre><code>
[ ] Regression 
[x] Bug report
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Expected behavior

If `registerQueueAsync` used with `useClass` in this way:

```typescript
    BullModule.registerQueueAsync({
      name: LINK_GAMES_QUEUE,
      useClass: QueueConfigService,
    }),
```

where QueueConfigService looks like

```typescript
import { BullModuleOptions, BullOptionsFactory } from '@nestjs/bull';
import { ConfigService } from '@nestjs/config';
import { Injectable } from '@nestjs/common';
import { LINK_GAMES_QUEUE, QUEUE_REDIS_DB } from '@app/constants';

@Injectable()
export class QueueConfigService implements BullOptionsFactory {
  constructor(private configService: ConfigService) {}

  createBullOptions(): BullModuleOptions {
    return {
      redis: {
        host: this.configService.get('REDIS_CACHE_HOST'),
        port: this.configService.get('REDIS_CACHE_PORT'),
        db: QUEUE_REDIS_DB,
      },
    };
  }
}
```

queue created as `default`.

**Comment by frankhock** [comment]
@solshark I ran into the same issue but using `useFactory`, you have to pass the `name` prop as part of the return object from your `createBullOptions ` method

**Comment by kamilmysliwiec** [comment]
Fixed in 0.1.1
